apps/src/code-studio/assets/SpriteLabAnimationUpload.jsx

```diff
@@ -0,0 +1,590 @@
+import React from 'react';
+import color from '@cdo/apps/util/color';
+import {makeEnum} from '@cdo/apps/utils';
+import ValidationStep, {Status} from '../../lib/ui/ValidationStep';
+import {
+  getManifest,
+  getLevelAnimationsFiles,
+  uploadAnimationToAnimationLibrary,
+  uploadMetadataToAnimationLibrary
+} from '@cdo/apps/assetManagement/animationLibraryApi';
+
+const EXTENSION_CHECK = 'extensionError';
+const FILENAME_CHECK = 'filenameError';
+const FILESIZE_CHECK = 'filesizeError';
+const ANIMATION_EXISTS_CHECK = 'animationExistsError';
+const AnimationLocations = makeEnum('library', 'level');
+
+const initialSpriteLabLibrariesState = {
+  spriteLabLevelAnimations: [],
+  spriteLabLibraryCategories: [],
+  spriteLabLibraryAnimations: []
+};
+
+const newImageState = {
+  fileData: null,
+  filename: '',
+  filePreviewURL: ''
+};
+
+const imageChecksInitialState = {
+  [EXTENSION_CHECK]: Status.WAITING,
+  [FILENAME_CHECK]: Status.WAITING,
+  [FILESIZE_CHECK]: Status.WAITING,
+  [ANIMATION_EXISTS_CHECK]: Status.WAITING
+};
+
+const initialUserInputMetaData = {
+  availability: '',
+  category: '',
+  aliases: []
+};
+
+const initialGeneratedMetadata = {
+  frameSize: {x: NaN, y: NaN},
+  destination: '',
+  metadata: ''
+};
+
+const initialComponentState = {
+  ...newImageState,
+  ...imageChecksInitialState,
+  ...initialGeneratedMetadata
+};
+
+const maxSize = 800;
+const minSize = 100;
+
+// Levelbuilder tool for adding animations to the Spritelab animation library.
+export default class AnimationUpload extends React.Component {
+  state = {
+    ...initialSpriteLabLibrariesState,
+    ...initialUserInputMetaData,
+    ...initialComponentState,
+    uploadStatus: Status.WAITING,
+    errorMessage: ''
+  };
+  componentDidMount() {
+    // Get list of animations from level-specific folder
+    getLevelAnimationsFiles().then(files => {
```

Comment:
Because of the delay here, we might want to considering adding 'loading' spinner? Not sure if that would improve the UX or draw more attention to the delay?

Reply:
Ahhh, whoops, I see you already did that! Good call!